### ISSUE TITLE
Avoid inconvenient behavior of BoxGymObsSpace and BoxGymActSpace during normalization

### ISSUE THREAD

**Issue by EBoguslawski** [rte-france/Grid2Op#446]
## Environment
 - Grid2op version: `1.8.2.dev4`
 - System: `Ubuntu 20.04`
 - Gym version: `0.21.0`

## Bug description
Hello,
I want to create two identical `BoxGymObsSpace` objects which I called in my code `gym_observation_space_1` and `gym_observation_space_2`.
When I use the `normalize_attr` method on `gym_observation_space_1`, it also impacts `gym_observation_space_2`, which is undesirable. 

## How to reproduce

Execute the following code

```python
import grid2op
from grid2op.gym_compat import BoxGymActSpace, BoxGymObsSpace
import numpy as np

env = grid2op.make("l2rpn_case14_sandbox")

divide = {"hour_of_day": np.ones(1)}

gym_observation_space_1 =  BoxGymObsSpace(env.observation_space,
                                        attr_to_keep=["curtailment_mw", "hour_of_day"],
                                        divide = divide
                                        )

gym_observation_space_2 =  BoxGymObsSpace(env.observation_space.copy(),
                                        attr_to_keep=["curtailment_mw", "hour_of_day"],
                                        divide = divide
                                        )


print("During normalization, attributes of gym_observation_space_1 will be divided by:", gym_observation_space_1._divide)
print("During normalization, attributes of gym_observation_space_2 will be divided by:", gym_observation_space_2._divide)

print("I normalize the 'curtailment_mw' attr of gym_obseration_space_1")
gym_observation_space_1.normalize_attr("curtailment_mw")
print("During normalization, attributes of gym_observation_space_1 will be divided by:", gym_observation_space_1._divide)
print("During normalization, attributes of gym_observation_space_2 will be divided by:", gym_observation_space_2._divide)
```

## Current output

```
During normalization, attributes of gym_observation_space_1 will be divided by: {'hour_of_day': array([1.])}
During normalization, attributes of gym_observation_space_2 will be divided by: {'hour_of_day': array([1.])}
I normalize the 'curtailment_mw' attr of gym_obseration_space_1
During normalization, attributes of gym_observation_space_1 will be divided by: {'hour_of_day': array([1.]), 'curtailment_mw': array([140., 120.,  70.,  70.,  40., 100.], dtype=float32)}
# HERE ############################
During normalization, attributes of gym_observation_space_2 will be divided by: {'hour_of_day': array([1.]), 'curtailment_mw': array([140., 120.,  70.,  70.,  40., 100.], dtype=float32)}
##################################
```
## Expected output

```
During normalization, attributes of gym_observation_space_1 will be divided by: {'hour_of_day': array([1.])}
During normalization, attributes of gym_observation_space_2 will be divided by: {'hour_of_day': array([1.])}
I normalize the 'curtailment_mw' attr of gym_obseration_space_1
During normalization, attributes of gym_observation_space_1 will be divided by: {'hour_of_day': array([1.]), 'curtailment_mw': array([140., 120.,  70.,  70.,  40., 100.], dtype=float32)}
# HERE ############################
During normalization, attributes of gym_observation_space_2 will be divided by: {'hour_of_day': array([1.])}
##################################
```

## Identified problem

It appears to be linked to the mutability of the dictionary `divide`. For example, the following code is working well:

```python
import grid2op
from grid2op.gym_compat import BoxGymActSpace, BoxGymObsSpace
import numpy as np

env = grid2op.make("l2rpn_case14_sandbox")

divide = {"hour_of_day": np.ones(1)}

gym_observation_space_1 =  BoxGymObsSpace(env.observation_space,
                                        attr_to_keep=["curtailment_mw", "hour_of_day"],
                                        divide = divide
                                        )

gym_observation_space_2 =  BoxGymObsSpace(env.observation_space.copy(),
                                        attr_to_keep=["curtailment_mw", "hour_of_day"],
                                       # HERE ###########
                                        divide = divide.copy()     # We need to copy the dictionary divide
                                       ###########
                                        )

```

To avoid such behavior, I suggest doing the `.copy()` directly in the code, in the file Grid2Op/blob/master/grid2op/gym_compat/box_gym_obsspace.py, line 550:

I would change it like this:
``` python
if subtract is None:
     subtract = {}
self._subtract = subtract # remplace with subtract.copy()
if divide is None:
     divide = {}
self._divide = divide # remplace with divide.copy()
```

**The suggestion also holds for the `_multiply` and `_add` attributes of the `BoxGymActSpace` class in `gym_compat`.**

Another drawback that arises by not doing a ".copy()" is the following: if you modify the `divide` dictionary a posteriori, you also modify `gym_observation_space_1._divide` with it.

**Comment by BDonnot** [comment]
Hello,

Thanks for spotting it. Can you do the fix and make a pull request?

- fork grid2op
- clone your fork (`git clone https://github.com/EBoguslawski/grid2op.git grid2op_dev`) 
- synch your local clone of the fork with branch dev_1.8.2 (`git add upstream https://github.com/rte-france/grid2op.git	; git fetch --all`) 
- make a branch from this (`git checkout -b fix_446 upstream/dev_1.8.2`)
- implement the fix you propose
- push your modifications `git push origin fix_446`
- make the pull request from your repo to the branch dev_1.8.2 of Rte grid2op

Thanks

(git commands given as example, most should work there might be typo)

**Comment by BDonnot** [comment]
Hi,

Has this been fixed now ?

**Comment by EBoguslawski** [comment]
Sorry, I totally forgot. I try to do it today.

**Comment by EBoguslawski** [comment]
Fixed with pull request #516

**Comment by BDonnot** [comment]
Thanks, I just merged your fix :-)
